### PR TITLE
Add cached search functionality to users

### PR DESCRIPTION
**Pull request by khalifan-kfan**
# Description

Add cached search functionality to the users admin screen 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Trello Ticket ID

https://trello.com/c/ORw6VvGn

## How Can This Been Tested?

Login as admin on this branch and try to search through users

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots

Screenshot 2023-06-29 at 11 36 27

### CODE REVIEW

**Review by rhodinemma — APPROVED** [review]
Thank you for this @khalifan-kfan

File: src/components/UserAccounts/index.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const UserAccounts = () => {
   const [creditDescription, setCreditDescription] = useState("");
   const [selectedUser, setSelectedUser] = useState("");
   const [word, setWord] = useState("");
-  const [searchList, setSearchList] = useState([]);
+  // const [searchList, setSearchList] = useState([]);
```
Comment (jnsereko) [inline src/components/UserAccounts/index.js:37]: why not remove this whole line

File: src/components/UserAccounts/index.js [hunk 2]
```diff
@@ -61,24 +61,22 @@ const UserAccounts = () => {
...
   const handleCallbackSearchword = ({ target }) => {
     const { value } = target;
     setWord(value);
     if (value !== "") {
-      searchThroughAccounts();
+      searchThroughAccounts(value);
     }
     if (value === "") {
-      setSearchList([]);
+      // setSearchList([]);
```
Comment (jnsereko) [inline src/components/UserAccounts/index.js:77]: same here

File: src/redux/actions/users.js [hunk 3]
```diff
@@ -22,14 +31,51 @@ export const getUsersFail = (error) => ({
...
-    const response = await axios.get(`/users?page=${page}`);
-    return dispatch(getUsersSuccess(response));
-  } catch (error) {
-    dispatch(getUsersFail(error));
+  // try {
+  //   const response = await axios.get(`/users?page=${page}`);
+  //   return dispatch(getUsersSuccess(response));
+  // } catch (error) {
+  //   dispatch(getUsersFail(error));
+  // }
```
Comment (jnsereko) [inline src/redux/actions/users.js:41]: same here

**Review by jnsereko — COMMENTED** [review]
great work @khalifan-kfan 
However, i ideally think it is not a good practice to have comment code in the code-base
cc @neelxie

**Review by neelxie — APPROVED** [review]
LGTM

**Pull request merged**